src/main/java/com/google/sps/data/Room.java

```diff
@@ -0,0 +1,55 @@
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.HashMap;
+import javax.servlet.http.HttpServletRequest;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+
+public class Room {
+
+    private ArrayList<String> members;
+    private LinkedList<Message> messages;
+    private ArrayList<String> video_urls;
+    private RoomState state;
+
+    /**
+      * Room constructor
+      * @param obj a JsonObject representing the configurations the Room creator specified
+      * @param urls an arraylist of the video urls of the initial Room playlist
+      * @return a new Room object
+      */
+    public Room(JsonObject obj, ArrayList<String> urls){
+        this.messages = new LinkedList<Message>();
+        this.state = new RoomState();
+        this.members = DataUtil.PARSER.fromJson(obj.getAsJsonArray("members"),ArrayList<String>.class);
+        this.video_urls = taurls;
+    }
+    
+    //Function that transforms a Room to a HashMap
```

Comment:
Don't understand why do you need a hashmap like this. You can simply convert a class to string or json if your goal is to convert to json later. 

No need manually build a map. are those fields as property of room class?